packages/trackerless-network/src/logic/RandomGraphNode.ts

```diff
@@ -0,0 +1,297 @@
+import { EventEmitter } from 'eventemitter3'
+import {
+    DhtNode,
+    PeerDescriptor,
+    DhtPeer,
+    ListeningRpcCommunicator,
+    ITransport,
+    ConnectionLocker,
+    keyFromPeerDescriptor
+} from '@streamr/dht'
+import {
+    StreamMessage,
+    LeaveStreamNotice,
+    MessageRef,
+    StreamMessageType,
+    GroupKeyRequest
+} from '../proto/packages/trackerless-network/protos/NetworkRpc'
+import { PeerList } from './PeerList'
+import { NetworkRpcClient } from '../proto/packages/trackerless-network/protos/NetworkRpc.client'
+import { RemoteRandomGraphNode } from './RemoteRandomGraphNode'
+import { INetworkRpc } from '../proto/packages/trackerless-network/protos/NetworkRpc.server'
+import { DuplicateMessageDetector } from './DuplicateMessageDetector'
+import { Logger } from '@streamr/utils'
+import { toProtoRpcClient } from '@streamr/proto-rpc'
+import { IHandshaker } from './neighbor-discovery/Handshaker'
+import { Propagation } from './propagation/Propagation'
+import { INeighborFinder } from './neighbor-discovery/NeighborFinder'
+import { INeighborUpdateManager } from './neighbor-discovery/NeighborUpdateManager'
+import { PeerIDKey } from '@streamr/dht/dist/src/helpers/PeerID'
+import { StreamNodeServer } from './StreamNodeServer'
+import { IStreamNode } from './IStreamNode'
+import { ProxyStreamConnectionServer } from './proxy/ProxyStreamConnectionServer'
+import { markAndCheckDuplicate } from './utils'
+
+export interface Events {
+    message: (message: StreamMessage) => void
+    targetNeighborConnected: (stringId: string) => void
+    nearbyContactPoolIdAdded: () => void
+}
+
+export interface StrictRandomGraphNodeConfig {
+    randomGraphId: string
+    layer1: DhtNode
+    P2PTransport: ITransport
+    connectionLocker: ConnectionLocker
+    ownPeerDescriptor: PeerDescriptor
+    peerViewSize: number
+    nearbyContactPool: PeerList
+    randomContactPool: PeerList
+    targetNeighbors: PeerList
+    handshaker: IHandshaker
+    neighborFinder: INeighborFinder
+    neighborUpdateManager: INeighborUpdateManager
+    propagation: Propagation
+    rpcCommunicator: ListeningRpcCommunicator
+    numOfTargetNeighbors: number
+    maxNumberOfContacts: number
+    minPropagationTargets: number
+    nodeName: string
```

Comment:
```suggestion
    name: string
```

No need for prefix as these are configs for a node.